ddtrace/opentelemetry/span_test.go

```diff
@@ -79,148 +77,134 @@ func TestSpanSetName(t *testing.T) {
 	tracer.Flush()
 	p, err := waitForPayload(ctx, payloads)
 	if err != nil {
-		t.Log(err)
-		t.Fail()
+		t.Fatalf(err.Error())
 	}
 	assert.Contains(p, "NewName")
 }
 
 func TestSpanEnd(t *testing.T) {
 	assert := assert.New(t)
-	testData := []struct {
-		trueName        string
-		falseName       string
-		trueError       codes.Code
-		trueErrorMsg    string
-		falseError      codes.Code
-		falseErrorMsg   string
-		trueAttributes  map[string]string
-		falseAttributes map[string]string
-	}{
-		{
-			trueName:        "trueName",
-			falseName:       "invalidName",
-			trueError:       codes.Error,
-			trueErrorMsg:    "error_description",
-			falseError:      codes.Ok,
-			falseErrorMsg:   "ok_description",
-			trueAttributes:  map[string]string{"trueKey": "trueVal"},
-			falseAttributes: map[string]string{"trueKey": "fakeVal", "invalidKey": "invalidVal"},
-		},
-	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
 	_, payloads, cleanup := mockTracerProvider(t)
 	tr := otel.Tracer("")
 	defer cleanup()
 
-	for _, test := range testData {
-		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-		_, sp := tr.Start(context.Background(), test.trueName)
-		sp.SetStatus(codes.Error, test.trueErrorMsg)
-		for k, v := range test.trueAttributes {
-			sp.SetAttributes(attribute.String(k, v))
-		}
-		assert.True(sp.IsRecording())
-		sp.End()
-		assert.False(sp.IsRecording())
-		// following operations should not be able to modify the Span
-		sp.SetName(test.trueName)
-		sp.SetStatus(test.falseError, test.falseErrorMsg)
-		for k, v := range test.trueAttributes {
-			sp.SetAttributes(attribute.String(k, v))
-			sp.SetAttributes(attribute.String(k, v))
-		}
+	name, ignoredName := "trueName", "invalidName"
+	code, codeMsg := codes.Error, "error_desc"
+	ignoredCode, ignoredMsg := codes.Ok, "ok_desc"
```

Comment:
Just a nit that sometimes you pair by type (e.g. name) and sometimes you pair by whether or not it's ignored (e.g. code, codeMsg). Doesn't matter which way, but we should be consistent if you want to do that.